integrations/spark/src/main/java/marquez/spark/agent/lifecycle/plan/InputDatasetVisitors.java

```diff
@@ -0,0 +1,34 @@
+package marquez.spark.agent.lifecycle.plan;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Supplier;
+import marquez.spark.agent.client.LineageEvent.Dataset;
+import org.apache.spark.sql.SQLContext;
+import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan;
+import scala.PartialFunction;
+
+/**
+ * Constructs a list of valid {@link LogicalPlan} visitors that can extract an input {@link
+ * Dataset}. Checks the classpath for classes that are not bundled with Spark to avoid {@link
+ * ClassNotFoundException}s during plan traversal.
+ */
+public class InputDatasetVisitors
+    implements Supplier<List<PartialFunction<LogicalPlan, List<Dataset>>>> {
+  private final SQLContext sqlContext;
+
+  public InputDatasetVisitors(SQLContext sqlContext) {
+    this.sqlContext = sqlContext;
+  }
+
+  @Override
+  public List<PartialFunction<LogicalPlan, List<Dataset>>> get() {
+    List<PartialFunction<LogicalPlan, List<Dataset>>> list = new ArrayList<>();
+    list.add(new HadoopFsRelationVisitor(sqlContext.sparkContext()));
+    list.add(new JDBCRelationVisitor(sqlContext));
+    if (BigQueryNodeVisitor.hasBigQueryClasses()) {
```

Comment:
Can we call the static method if the class itself depends on classes missing in the classpath? Or would an exception occured only when calling the method actually depending on it?

Reply:
is there a way to test this in the build?

Reply:
The class itself doesn't have a direct dependency on the BigQueryRelation class- only if the class had a static initializer (static property or static initialization block) that referenced the BigQueryRelation class would it cause a problem.

Reply:
👍